cmd/recovereth/main.go

```diff
@@ -0,0 +1,396 @@
+
+		if err := json.Unmarshal([]byte(keyjson), k); err != nil {
+			return err
+		}
+		addr := common.HexToAddress(k.Address)
+
+		k2, err := flipBit(k.Crypto, i)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("C  %s -> %s \n", k.Crypto.CipherText, k2.CipherText)
+		fmt.Printf("IV %s -> %s \n", k.Crypto.CipherParams.IV, k2.CipherParams.IV)
+		fmt.Printf("Sa %s -> %s \n", k.Crypto.KDFParams["salt"].(string), k2.KDFParams["salt"].(string))
+
+		keyBytes, err := decryptKeyV3x(k2, auth)
+		if err != nil {
+			return err
+		}
+		key := crypto.ToECDSAUnsafe(keyBytes)
+		if err != nil {
+			return err
+		}
+		recovered := crypto.PubkeyToAddress(key.PublicKey)
+		fmt.Printf("found address %s (looking for %s)\n", recovered.Hex(), addr.Hex())
+		if recovered == addr {
+			fmt.Printf("Recovery successfull!\n")
+			json.Marshal(k2)
+			json, _ := json.MarshalIndent(k2, "", "    ")
+			fmt.Printf("Recovered json-file: \n------------\n%v\n------------\n", string(json))
+			fmt.Printf("Please report to the go-ethereum repository "+
+				"that you were able to recover a wallet by 'bitflip'-manipulation at bit %d.", i)
+			return nil
+		}
+	}
+}
+
+func testDerivedKeyBitflip(keyjson, auth string) error {
+	k := new(encryptedKeyJSONV3)
+	if err := json.Unmarshal([]byte(keyjson), k); err != nil {
+		log.Fatal(err)
+	}
+	addr := common.HexToAddress(k.Address)
+	iv, _ := hex.DecodeString(k.Crypto.CipherParams.IV)
+	cipherText, _ := hex.DecodeString(k.Crypto.CipherText)
+	derivedKey, _ := getKDFKey(k.Crypto, auth)
+
+	derivedKeyHex := common.Bytes2Hex(derivedKey)
+	fmt.Printf("Derived key %s\n", derivedKeyHex)
+	var i uint
+	for i = 0; i < 256; i++ {
+		corruptedKeyHex := flipBitInString(i, derivedKeyHex)
+		corruptedKey := common.Hex2Bytes(corruptedKeyHex)
+		plainText, err := aesCTRXOR(corruptedKey[:16], cipherText, iv)
+		key := crypto.ToECDSAUnsafe(plainText)
+
+		if err != nil {
+			return err
+		}
+		recovered := crypto.PubkeyToAddress(key.PublicKey)
+		fmt.Printf("found address %s (looking for %s)\n", recovered.Hex(), addr.Hex())
+		if recovered == addr {
+			fmt.Printf("Recovery successfull!\n")
+			fmt.Printf("Recovered privatekey: \n------------\n0x%x\n------------\n", plainText)
+			fmt.Printf("Please report to the go-ethereum repository "+
+				"that you were able to recover a wallet by 'derived-key bitflip-manipulation' at bit %d.", i)
+			return nil
+		}
+	}
+	return errors.New("Exhausted")
+}
+
+func main() {
+	if err := app.Run(os.Args); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+// confirm displays a text and asks for user confirmation
+func confirm(text string) bool {
+	fmt.Printf(text)
+	fmt.Printf("\nEnter 'ok' to proceed:\n>")
+
+	text, err := bufio.NewReader(os.Stdin).ReadString('\n')
+	if err != nil {
+		log.Fatalf("Failed to read user input", "err", err)
+	}
+
+	if text := strings.TrimSpace(text); text == "ok" {
+		return true
+	}
+	return false
+}
+
+
+func recover(ctx *cli.Context) error {
```

Comment:
In this function you use `utils.Fatalf`. I think it's better to use `log.Fatalf` to be consistent in coding style